src/NServiceBus.Core/Pipeline/ConnectorContextExtensions.cs

```diff
@@ -159,6 +215,10 @@ public static IAuditContext CreateAuditContext(this ForkConnector<IIncomingPhysi
         /// </summary>
         public static IForwardingContext CreateForwardingContext(this ForkConnector<IIncomingPhysicalMessageContext, IForwardingContext> forwardingContext, OutgoingMessage message, string forwardingAddress, IIncomingPhysicalMessageContext sourceContext)
         {
+            Guard.AgainstNull(nameof(message), message);
+            Guard.AgainstNull(nameof(forwardingAddress), forwardingAddress);
```

Comment:
I've changed it to `Guard.AgainstNullAndEmpty()` and I've repeated the constraints in `ForwardingContext`.

Note that the constraints still have to be present here because the parameter names are different, so relying on the `ForwardingContext` to guard would result in confusing exception messages for the caller. (It probably wouldn't be *hugely* confusing, but nevertheless the exception messages would be using parameter names only present on an internal class method).